test/support/test_complex.h

```diff
@@ -86,14 +86,27 @@ run_test()
 //         // ...
 //     }
 #define IF_DOUBLE_SUPPORT(x)                                                                          \
-    if constexpr (HasDoubleSupportInRuntime::value) { x; }
+    TestUtils::invoke_test_if<>(HasDoubleSupportInRuntime(), [](){ x; });
 
 // We should use this macros to avoid compile-time error in code with long double type in Kernel.
 #define IF_LONG_DOUBLE_SUPPORT(x)                                                                     \
-    if constexpr (HasLongDoubleSupportInCompiletime::value) { x; }
+    TestUtils::invoke_test_if<>(HasLongDoubleSupportInCompiletime(), []() { x; });
```

Comment:
These "syntax sugar" macros seem rather useless and not consistently used even within this patch. I would suggest to remove those.

Reply:
Fixed (these macros was removes).

Reply:
I prepared this fix but there are too many changes.
My suggestion do not try to fix this point in this PR and discuss it separatelly.